tests/apollo/test_skvbc_reconfiguration.py

```diff
@@ -555,6 +555,228 @@ async def test_addRemove_command(self, bft_network):
+             The test does the following:
+             1. A client sends a add node command which will also wedge the system on next next checkpoint
+             2. Validate that all replicas have stopped
+             3. Load a new configuration to the bft network
+             4. Add node is done in phases, (n=4,f=1,c=0)->(n=6,f=1,c=0)->(n=7,f=2,c=0)
+                Note: For new replicas to catch up with exiting replicas through ST, existing replicas must
+                      move the checkpoint window, that means for n=7 configuration, there must be 5 non-faulty 
+                      replicas to move the checkpoint window, hence new replicas are added in two phases
+             5. Rerun the cluster with only new configuration and make sure they succeed to perform transactions in fast path
+         """
+        conf = TestConfig(n=4,
+                   f=1,
+                   c=0,
+                   num_clients=10,
+                   key_file_prefix=KEY_FILE_PREFIX,
+                   start_replica_cmd=start_replica_cmd,
+                   stop_replica_cmd=None,
+                   num_ro_replicas=0)
+        await bft_network.change_configuration(conf)
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+        for i in range(100):
+            await skvbc.write_known_kv()
+        client = bft_network.random_client()
+        client.config._replace(req_timeout_milli=10000)
+        checkpoint_before = await bft_network.wait_for_checkpoint(replica_id=0)
+        op = operator.Operator(bft_network.config, client,  bft_network.builddir)
+        test_config = 'new_configuration_n_6_f_1_c_0'
+        await op.add_remove_with_wedge(test_config)
+        await self.verify_replicas_are_in_wedged_checkpoint(bft_network, checkpoint_before, range(bft_network.config.n))
+        await self.verify_last_executed_seq_num(bft_network, checkpoint_before)
+        await self.validate_stop_on_super_stable_checkpoint(bft_network, skvbc)
+        bft_network.stop_all_replicas()
+        # We now expect the replicas to start with a fresh new configuration which means that we
+        # need to see in the logs that isNewStorage() = true. Also,
+        # we expect to see that lastStableSeqNum = 0 (for example)
+        conf = TestConfig(n=6,
+                   f=1,
+                   c=0,
+                   num_clients=10,
+                   key_file_prefix=KEY_FILE_PREFIX,
+                   start_replica_cmd=start_replica_cmd,
+                   stop_replica_cmd=None,
+                   num_ro_replicas=0)
+        await bft_network.change_configuration(conf)
+        initial_prim = 0
+        new_replicas = {4, 5}
+        on_time_replicas = bft_network.all_replicas(without=new_replicas)
+        bft_network.start_replicas(on_time_replicas)
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+        for i in range(151):
```

Comment:
We have to issue 151 requests after the restart to initiate state transfer, right? (because we remove the metadata)

Reply:
yes 151 requests are issues to initiate state transfer